src/creature.cpp

```diff
@@ -951,7 +951,7 @@ void Creature::deal_damage_handle_type( const effect_source &source, const damag
         case damage_type::STAB:
         case damage_type::BULLET:
             // these are bleed inducing damage types
-            if( is_avatar() || is_npc() ) {
+            if( ( is_avatar() || is_npc() ) && ( as_character()->is_bp_armored( bp ) ? !one_in( 3 ) : true ) ) {
```

Comment:
Instead of this check wouldn't it be better to implement a bleed resistance attribute for CBMs? This way ther ewould be no hardcoded one_in(3) probability and you'd know how much portection the CBM gives directly from the json.